spikeinterface/toolkit/preprocessing/deepinterpolation.py

```diff
@@ -54,11 +54,37 @@ def __init__(self, recording_segment, model,
         self.local_std = local_std
         
     def get_traces(self, start_frame, end_frame, channel_indices):        
+        
+        n_frames = self.parent_recording_segment.get_num_samples()
 
+        if start_frame==None:
+            start_frame = 0
+        
+        if end_frame==None:
+            end_frame=n_frames
+
+        # only apply DI to frames that have full training data
+        # for those that don't, just return uninterpolated data
+        if start_frame<self.pre_frames+self.pre_post_omission:
+            true_start_frame = self.pre_frames+self.pre_post_omission
+            array_to_append_front = self.parent_recording_segment.get_traces(start_frame=0,
+                                                                             end_frame=true_start_frame,
+                                                                             channel_indices=None)
+        else:
+            true_start_frame = start_frame
+            
+        if end_frame>n_frames-self.post_frames-self.pre_post_omission:
+            true_end_frame = n_frames-self.post_frames-self.pre_post_omission
+            array_to_append_back = self.parent_recording_segment.get_traces(start_frame=true_end_frame,
+                                                                            end_frame=n_frames,
+                                                                            channel_indices=None)
```

Comment:
This look like a margin handling.
We have a function for this already.
`spikeinterfacea.toolkit.preprocessing.get_chunk_with_margin()`